src/components/provider/component_defaults/component_defaults.test.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FunctionComponent, PropsWithChildren } from 'react';
+import { renderHook } from '../../../test/rtl';
+
+import {
+  EuiComponentDefaultsProvider,
+  useComponentDefaults,
+  usePropsWithComponentDefaults,
+} from './component_defaults';
+
+describe('EuiComponentDefaultsProvider', () => {
+  describe('useComponentDefaults', () => {
+    it('allows accessing provided `componentDefaults` from anywhere', () => {
+      const wrapper: FunctionComponent<PropsWithChildren> = ({ children }) => (
+        <EuiComponentDefaultsProvider
+          componentDefaults={{
+            EuiPortal: {
+              insert: {
+                sibling: document.createElement('div'),
+                position: 'before',
+              },
+            },
+          }}
+        >
+          {children}
+        </EuiComponentDefaultsProvider>
+      );
+      const { result } = renderHook(useComponentDefaults, { wrapper });
+
+      expect(result.current).toMatchInlineSnapshot(`
+        Object {
+          "EuiPortal": Object {
+            "insert": Object {
+              "position": "before",
+              "sibling": <div />,
+            },
+          },
+        }
+      `);
```

Comment:
What do you think of using object equality matcher here instead?
```suggestion
      expect(result.current).toEqual({
        EuiPortal: {
          insert: {
            position: 'before',
            sibling: expect.any(HTMLDivElement),
          },
        },
      });
```

Reply:
I'm good with that! I'm always so lazy about writing out object assertions vs snapshotting objects 🤣